app.py

```diff
@@ -49,7 +49,14 @@ def _list_pulp_python_repositories(instance_host: str, *, username: str, passwor
         simple_repository = (
             f"{result['base_url']}simple" if result["base_url"].endswith("/") else f"{result['base_url']}/simple"
         )
-        yield urlunparse(("https", instance_host, simple_repository, "", "", ""))
+
+        index = -1
+        for i in range(0, 3):
+            index = simple_repository.find("/", index + 1)
+        if instance_host[-1:] == "/":
+            yield urlunparse(("https", instance_host, simple_repository[index + 1 :], "", "", ""))
+        else:
+            yield urlunparse(("https", instance_host, simple_repository[index:], "", "", ""))
```

Comment:
It looks like the issue is caused by wrong parsing of `base_url` and joining it with the `instance_host` which in turn does something like:

```python
"https://pulp.operate-first.cloud/" + "https://pulp-api-5c5f4669d8-qgvvh/pypi/hello-world/simple"
```

What about changing the code to something like this?

```diff
diff --git a/app.py b/app.py
index bfe20d7..c8fde04 100755
--- a/app.py
+++ b/app.py
@@ -19,6 +19,7 @@
 
 import click
 import logging
+from urllib.parse import urlparse
 from urllib.parse import urlunparse
 from typing import Generator
 
@@ -46,10 +47,10 @@ def _list_pulp_python_repositories(instance_host: str, *, username: str, passwor
     response.raise_for_status()
 
     for result in response.json()["results"]:
-        simple_repository = (
-            f"{result['base_url']}simple" if result["base_url"].endswith("/") else f"{result['base_url']}/simple"
+        path = urlparse(result["base_url"]).path
+        yield urlunparse(
+            ("https", instance_host, f"{path}/simple" if not path.endswith("/") else f"{path}simple", "", "", "")
         )
-        yield urlunparse(("https", instance_host, simple_repository, "", "", ""))
 
 
 @click.command()
```

Reply:
I like the idea of parsing the URL and then extracting the path, it seems like a more stable solution rather than finding the third indexed `/` in a URL.